src/lesson-3/modules/tasks/task.service.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  Bug,
+  Epic,
+  FilterParams,
+  NewTaskBody,
+  Story,
+  Subtask,
+  TaskDetails,
+  TaskType,
+  UnitedSchema,
+} from './task.types';
+
+export class TaskService {
+  #tasks: UnitedSchema[];
+
+  constructor(tasks: UnitedSchema[]) {
+    this.#tasks = tasks;
+  }
+
+  getAll() {
+    return this.#tasks;
+  }
+
+  // create new Task
+  createNewInstance({
+    type,
+    taskDetail,
+  }: {
+    type: TaskType;
+    taskDetail: NewTaskBody;
+  }) {
+    let task!: UnitedSchema;
+
+    if (type === 'bug') {
+      task = new Bug({ ...taskDetail, createdAt: new Date() });
+    }
+
+    if (type === 'story') {
+      task = new Story({ ...taskDetail, createdAt: new Date() });
+    }
+
+    if (type === 'epic') {
+      task = new Epic({ ...taskDetail, createdAt: new Date() });
+    }
+
+    if (type === 'subtask') {
+      task = new Subtask({ ...taskDetail, createdAt: new Date() });
+    }
+
+    this.#tasks.push(task);
+    return task;
+  }
+
+  // get by id
+  getTaskById = (taskId: string): UnitedSchema | null => {
+    const find = this.#tasks.find((item) => item.id === taskId);
+    return find || null;
```

Comment:
`null` не обовʼязково повертати якщо `find` може бути `undefined`
```suggestion
    return find
```

Reply:
моя професійна деформація))